namerd/docs/storage.md

```diff
@@ -23,19 +23,164 @@ Stores the dtab in memory.  Not suitable for production use.
+  ```
+  
+2. Define configuration of `Namerd` with k8s cluster storage.
+  * *experimental* -- Required. Has to be set to `true` as `k8s` storage is experimental feature right now.
+  * *host* -- Optional. The location of the Kubernetes API. (default: "kubernetes.default.svc.cluster.local")
+  * *port* -- Optional. The port used to connect to the Kubernetes API. (default: 443)
+  * *tls*  -- Optional. Whether to connect to the Kubernetes API using TLS. (default: true)
+  * *tlsWithoutValidation* -- Optional. Whether to disable certificate checking against the Kubernetes
   API. Meaningless if *tls* is false. (default: false)
-* *authTokenFile* -- Optional. The location of the token used to authenticate against the Kubernetes
+  * *authTokenFile* -- Optional. The location of the token used to authenticate against the Kubernetes
   API, if any. (default: no authentication)
-* *namespace* The Kubernetes namespace in which dtabs will be stored. This should usually be the
+  * *namespace* The Kubernetes namespace in which dtabs will be stored. This should usually be the
   same namespace in which namerd is running. (default: "default")
 
+*Example:*  
+```yaml
+  storage:
+    kind: io.l5d.k8s
+    experimental: true
+```
+
+*Complete example of `Namerd` configuration with `k8s` storage and exposed 2 serivces for sync with `Linkerd` and `Namerd API`:*  
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: namerd-sync
+spec:
+  selector:
+    app: namerd
+  ports:
+  - name: sync
+    port: 4100
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: namerd-api
+spec:
+  selector:
+    app: namerd
+  ports:
+  - name: api
+    port: 4180
+---
+metadata:
+  name: d-tab.l5d.io # the hyphen is required by the Kubernetes API. This will be converted to the CamelCase name "DTab".
+apiVersion: extensions/v1beta1
+kind: ThirdPartyResource
+description: stores dtabs used by Buoyant's `namerd` service
+versions:
+  - name: v1alpha1 # Do not change this value as it hardcoded in Namerd and doesn't work with other value.
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: namerd-config
+data:
+  config.yml: |-
+    admin:
+      port: 9991
+    storage:
+      kind: io.l5d.k8s
+      experimental: true
+    namers:
+      - kind: io.l5d.k8s
+        experimental: true
+        host: 127.0.0.1
+        port: 8001
+    interfaces:
+      - kind: io.l5d.thriftNameInterpreter
+        ip: 0.0.0.0
+        port: 4100
+      - kind: io.l5d.httpController
+        ip: 0.0.0.0
+        port: 4180
+---
+kind: ReplicationController
+apiVersion: v1
+metadata:
+  name: namerd
+spec:
+  replicas: 1
+  selector:
+    app: namerd
+  template:
+    metadata:
+      labels:
+        app: namerd
+    spec:
+      dnsPolicy: ClusterFirst
+      volumes:
+        - name: namerd-config
+          configMap:
+            name: namerd-config
+      containers:
+        - name: namerd
+          image: buoyantio/namerd:0.7.4
```

Comment:
Does anyone have suggestions wrt how we should deal with tags like this in our docs? We don't have a great way to keep these up to date...